homeassistant/components/acmeda/hub.py

```diff
@@ -0,0 +1,110 @@
+"""Code to handle a Pulse Hub."""
+import asyncio
+from typing import Optional
+
+import aiopulse
+
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import LOGGER
+from .errors import CannotConnect
+
+
+class PulseHub:
+    """Manages a single Pulse Hub."""
+
+    def __init__(self, hass, config_entry):
+        """Initialize the system."""
+        self.config_entry = config_entry
+        self.hass = hass
+        self.available = True
+        self.authorized = False
+        self.api: Optional[aiopulse.Hub] = None
+        self.parallel_updates_semaphore = None
+        self.task = None
+
+    @property
+    def host(self):
+        """Return the host of this hub."""
+        return self.config_entry.data["host"]
+
+    async def async_setup(self, tries=0):
+        """Set up a Pulse Hub based on host parameter."""
+        host = self.host
+        hass = self.hass
+
+        hub = aiopulse.Hub(host)
+
+        try:
+            # Create a Hub object and verify connection.
+            try:
+                # self.task = hass.async_create_task(hub.run())
+                self.task = asyncio.create_task(hub.run())
+                LOGGER.debug("Hub running")
+            except (aiopulse.InvalidResponseException):
+                raise CannotConnect
+
+            except (asyncio.TimeoutError):
+                raise CannotConnect
+
+        except CannotConnect:
+            LOGGER.error("Error connecting to the Pulse Hub at %s", host)
+            raise ConfigEntryNotReady
+
+        except Exception:  # pylint: disable=broad-except
+            LOGGER.exception("Unknown error connecting with Pulse Hub at %s", host)
+            return False
+
+        self.api = hub
+
+        hass.async_create_task(
```

Comment:
It might be helpful to keep track of this in a constant so you can use it for unload as well

```
    for platform in PLATFORMS:
        hass.async_create_task(
            hass.config_entries.async_forward_entry_setup(config_entry, platform)
        )
```